main.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+type ConfigSettings struct {
+	Port  string   `yaml:"port"`
+	Repos []string `yaml:"repos"`
+}
+
+// Global Configuration for the service
+var Config ConfigSettings
```

Comment:
Answer: I like the use of a public singleton here as it provides an authoritative source for the configuration of the service which can be referenced from external go code if necessary. This configuration is intended to be used from multiple sources in the service and so I believe it makes sense for it to be global. Packaging these fields in a struct helps to indicate that these fields are closely related in their purpose and clarifies that they are intended specifically as service configurations. After reviewing this code I believe a struct pointer would make more sense here as this will help prevent developers from inadvertently referencing stale copies of the configuration. Also, this allows Config to be a nil value making it easier to verify if the services configuration has been properly initialized.

Reply:
The reason we tend to avoid singletons is because of how brittle they can be. Having a global package level singleton works here because it's one file with simple functionality but what happens if it was a larger service? Every time this config is referenced you have to verify it has been initialized or initialize it yourself. The better solution 95% of the time is to have a struct that is initialized on startup that initializes the config and maintains that state within itself and then all handlers and functions that deal with that tightly coupled data are just methods on that struct.

Reply:
Thanks for the clarification, I've gone ahead and refactored the code to remove the global singleton to help make the code more robust.